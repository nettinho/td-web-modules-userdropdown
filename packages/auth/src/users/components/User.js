import _ from "lodash/fp";
import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { Header, Icon, Segment } from "semantic-ui-react";
import { connect } from "react-redux";
import { FormattedMessage } from "react-intl";
import { UserBreadcrumbs } from "./UserBreadcrumbs";
import UserAcls from "./UserAcls";
import UserGroupAcls from "./UserGroupAcls";

export const User = ({ user }) =>
  _.isEmpty(user) ? null : (
    <Fragment>
      <UserBreadcrumbs name={user.user_name} />
      <Segment>
        <Header as="h2">
          <Icon name="user" circular />
          <Header.Content>
            {user.user_name}
            <Header.Subheader>
              {user.is_admin ? (
                <FormattedMessage id="user.type.admin" />
              ) : (
                <FormattedMessage id="user.type.user" />
              )}
            </Header.Subheader>
          </Header.Content>
        </Header>
        {user.full_name}
        <br />
        {user.email}
        <br />
        {_.join(" ")(user.groups)}
        <br />
        <br />
        <br />
        <UserAcls />
        <UserGroupAcls />
      </Segment>
    </Fragment>
  );

User.propTypes = {
  user: PropTypes.object
};

const mapStateToProps = ({ user }) => ({ user });

export default connect(
  mapStateToProps,
  null
)(User);
