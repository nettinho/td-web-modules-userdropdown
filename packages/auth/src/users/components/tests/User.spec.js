import React from "react";
import { shallow } from "enzyme";
import { User } from "../User";

describe("<User />", () => {
  const user = {
    id: 1,
    user_name: "joe",
    is_admin: true,
    full_name: "Joe Bloggs",
    email: "joe@bloggs.net"
  };
  const props = { user };

  it("matches the latest snapshot", () => {
    const wrapper = shallow(<User {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
