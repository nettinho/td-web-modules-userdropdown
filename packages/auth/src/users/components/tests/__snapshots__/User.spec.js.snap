// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> matches the latest snapshot 1`] = `
<React.Fragment>
  <UserBreadcrumbs
    name="joe"
  />
  <Segment>
    <Header
      as="h2"
    >
      <Icon
        as="i"
        circular={true}
        name="user"
      />
      <HeaderContent>
        joe
        <HeaderSubheader>
          <FormattedMessage
            id="user.type.admin"
            values={Object {}}
          />
        </HeaderSubheader>
      </HeaderContent>
    </Header>
    Joe Bloggs
    <br />
    joe@bloggs.net
    <br />
    <br />
    <br />
    <br />
    <Connect(UserAcls) />
    <Connect(UserGroupAcls) />
  </Segment>
</React.Fragment>
`;
